plugins/inlinetoolbar/plugin.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or http://ckeditor.com/license
+ */
+
+( function() {
+	'use strict';
+
+	CKEDITOR.plugins.add( 'inlinetoolbar', {
+		requires: 'balloonpanel',
+		init: function( editor ) {
+			CKEDITOR.ui.inlineToolbarView.prototype = CKEDITOR.tools.extend( {}, CKEDITOR.ui.balloonPanel.prototype );
+			CKEDITOR.ui.inlineToolbarView.prototype.build = function() {
+				CKEDITOR.ui.balloonPanel.prototype.build.call( this );
+				this.parts.title.remove();
+				this.parts.close.remove();
+			};
+
+			CKEDITOR.ui.inlineToolbarView.prototype._getAlignments = function( elementRect, panelWidth, panelHeight ) {
+				var filter = [ 'top hcenter', 'bottom hcenter' ],
+					output = {},
+					alignments = CKEDITOR.ui.balloonPanel.prototype._getAlignments.call( this, elementRect, panelWidth, panelHeight );
+				for ( var a in alignments ) {
+					if ( CKEDITOR.tools.indexOf( filter, a ) !== -1 ) {
+						output[ a ] = alignments[ a ];
+					}
+				}
+				return output;
+			};
+
+			CKEDITOR.ui.inlineToolbarView.prototype._detachListiners = function() {
+				if ( this.listeners ) {
+					this.editor.removeListener( 'resize', this.listeners.resize );
+					this.listeners.scrollEvent.removeListener();
+					this.listeners = null;
+				}
+			};
+
+			CKEDITOR.ui.inlineToolbarView.prototype.destroy = function() {
+				CKEDITOR.ui.balloonPanel.prototype.destroy.call( this );
+				this._detachListiners();
+			};
+
+			CKEDITOR.ui.inlineToolbarView.prototype.create = function( element ) {
+				this.attach( element );
+
+				var that = this,
+					editable = this.editor.editable();
+				this._detachListiners();
+				this.listeners = {
```

Comment:
I don't like how currently listeners are working. If `attachListener` and `on` methods return object with `removeListener` method, then `listeners` could be just an array of such objects, which would simplify `_detachListeners` implementation to something like:

```javascript
CKEDITOR.tools.forEach( this.listeners, function( listener ) {
    listener.removeListener();
} );
```

It'll also make whole listener code more universal and flexible.